pam/internal/gdm/conversation.go

```diff
@@ -68,23 +68,24 @@ func SendData(pamMTx pam.ModuleTransaction, d *Data) (*Data, error) {
 	}
 
 	gdmData, err := NewDataFromJSON(jsonValue)
+	if err != nil {
+		log.Debugf(context.TODO(), "Failed parsing GDM result %q: %v", jsonValue, err)
+		return nil, err
```

Comment:
I would expect the caller to log the error we return here, so this would log the error twice (or, if the caller decides that the error should not be logged, then we also shouldn't log it here). Let's wrap the error instead and make sure that it's logged by the caller if we want errors returned by `SendData` to be logged:



```suggestion
		return nil, fmt.Errorf("Failed to parse GDM result %q: %w", jsonValue, err)
```

Reply:
Not sure about this, since `SendData` return error will eventually be sent back to Pam and so the interface will show it, and I feel we end up showing too much info, so I'm fine to wrap the error, but I'd avoid showing the json value there.

Reply:
> Not sure about this, since SendData return error will eventually be sent back to Pam and so the interface will show it

Sounds like we should not return the actual error to PAM but a more generic error message which is understandable for the user (something like "An unexpected error occurred during login") and log the actual error instead so that we or an admin can see it to debug the error.

Reply:
Yeah, I agree... But well, I wouldn't handle it here since it's many paths to change, and at the same time I don't want too generic not to make impossible to debug.

Reply:
Doesn't have to happen here, but then let's track it somewhere. And I would drop the `log.Debugf` statement here to avoid that we log the error twice once that's implemented.

Reply:
Well I can still add a TODO comment in that case, but I'm still unsure we should handle the logging at each call, as I feel the code becomes too cluttered for no much reason, given that these "low level" calls to Gdm have already all the context we may need (due to the content they send / receive).

Reply:
I've opened https://warthogs.atlassian.net/browse/UDENG-6197 to track this and added a todo comment, let's not block on this :)